### PR TITLE
feat: add option jump_type to lsp_definitions and lsp_implementations

### LINKED ISSUE #982: Is it possible for the builtin.lsp_definitions to default to a new tab?
When calling `builtin.lsp_definitions`, if there is only one result, the fuzzy picker is skipped and neovim navigates directly to the definition. Is there a way to make this single definition open in a new tab?

### PR DESCRIPTION
**Pull request by zeertzjq**
Closes #982.

### CODE REVIEW

**Comment by zeertzjq** [comment]
There is a typo

**Review by Conni2461 — APPROVED** [review]
LGTM thanks :)

**Comment by Conni2461** [comment]
> There is a typo

where? what?

File: lua/telescope/builtin/init.lua [hunk 1]
```diff
@@ -316,10 +316,12 @@ builtin.lsp_references = require("telescope.builtin.lsp").references
 
 --- Goto the definition of the word under the cursor, if there's only one, otherwise show all options in Telescope
 ---@param opts table: options to pass to the picker
+---@field jump_type string: how to go to definition if there is only one, values: "tab", "split", "vsplit", "never"
 builtin.lsp_definitions = require("telescope.builtin.lsp").definitions
 
 --- Goto the implementation of the word under the cursor if there's only one, otherwise show all options in Telescope
 ---@param opts table: options to pass to the picker
+---@field jump_type string: how to go to definition if there is only one, values: "tab", "split", "vsplit", "never"
```
Comment (zeertzjq) [inline lua/telescope/builtin/init.lua:324]: There is typo in this line, but fixing this will make the line too long.

Comment (Conni2461) [reply lua/telescope/builtin/init.lua:324]: i see

Comment (zeertzjq) [reply lua/telescope/builtin/init.lua:324]: Oh, it seems it won't make this line too long.

Comment (zeertzjq) [reply lua/telescope/builtin/init.lua]: Also, due to the line length limit, I couldn't add how to keep the original behavior to the documentation comments.

Comment (Conni2461) [reply lua/telescope/builtin/init.lua]: you meant that right

Comment (Conni2461) [reply lua/telescope/builtin/init.lua]: > Also, due to the line length limit, I couldn't add how to keep the original behavior to the documentation comments.

Just not define it. Yeah its currently a limitation we have to live with. Ill merge. Thanks for the PR :)

File: doc/telescope.txt [hunk 2]
```diff
@@ -798,6 +802,10 @@ builtin.lsp_implementations({opts})            *builtin.lsp_implementations()*
     Parameters: ~
         {opts} (table)  options to pass to the picker
 
+    Fields: ~
+        {jump_type} (string)  how to go to implementation if there is only one,
```
Comment (zeertzjq) [inline doc/telescope.txt:806]: Oh, this line seems to have 79 characters.

Comment (Conni2461) [reply doc/telescope.txt:806]: something is configured wrong in docgen. Ill fix at some point in time. Thanks

Comment (zeertzjq) [reply doc/telescope.txt:806]: Maybe "go to" can be changed to "goto".

**Pull request merged**